### PR TITLE
meas_astrom tickets/DM-2535: switch from compound fields to FunctorKeys

### PR DESCRIPTION
**Pull request by TallJimbo**

### CODE REVIEW

File: tests/testMatchOptimisticB.py [hunk 1]
```diff
@@ -132,8 +133,9 @@ def computePosRefCatalog(self, sourceCat):
         refCat = afwTable.SimpleCatalog(minimalPosRefSchema)
         for source in sourceCat:
             refObj = refCat.addNew()
-            refObj.set("coord", source.get("coord"))
-            refObj.set("centroid", source.getCentroid())
+            refObj.setCoord(source.getCoord())
+            refObj.set("centroid_x", source.getX())
+            refObj.set("centroid_y", source.getY())
```
Comment (r-owen) [inline tests/testMatchOptimisticB.py]: It is practical to set the two fields at once using a centroidKey? If so, that would be cleaner.

Comment (TallJimbo) [inline tests/testMatchOptimisticB.py]: I disagree that it's cleaner.  It's a lot of setup for a one-liner, and while it'd be more efficient if it was done repeatedly in production code, I don't think it matters in test code.

Comment (r-owen) [inline tests/testMatchOptimisticB.py]: Point taken. I will note that tests are often used as examples, and using keys is a better example. But as you say...it's a lot of extra code for no performance gain.

**Comment by TallJimbo** [comment]
Merged to master on tickets/DM-1766.

**Pull request closed without merging**